src/main/java/com/robertx22/mine_and_slash/capability/entity/ResourceTracker.java

```diff
@@ -0,0 +1,131 @@
+package com.robertx22.mine_and_slash.capability.entity;
+
+import com.robertx22.mine_and_slash.saveclasses.unit.ResourceType;
+
+/**
+ * Accumulates resource LOSS per type (spend, drains, damage, etc).
+ * Call addLoss(...) whenever a resource actually decreases.
+ * Use consumeThresholds(...) / addAndConsumeForKey(...) to fire effects and keep remainder.
+ */
+public class ResourceTracker {
```

Comment:
This tracks only loss, not gain, so let's add more context

```suggestion
public class ResourceLossTracker {
```

Reply:
To future proof I'd rather not change the name of the class here, unless we are to say we will never track gained resource (which I think would be a mistake).